src/txdb.cpp

```diff
@@ -332,19 +367,27 @@ bool CBlockTreeDB::LoadBlockIndexGuts(const Consensus::Params& consensusParams,
                 pindexNew->nTime          = diskindex.nTime;
                 pindexNew->nBits          = diskindex.nBits;
                 pindexNew->nNonce         = diskindex.nNonce;
-                pindexNew->proof          = diskindex.proof;
                 pindexNew->nStatus        = diskindex.nStatus;
                 pindexNew->nTx            = diskindex.nTx;
-                pindexNew->dynafed_params       = diskindex.dynafed_params;
-                pindexNew->m_signblock_witness = diskindex.m_signblock_witness;
-
-                const uint256 block_hash = pindexNew->GetBlockHash();
-                // Only validate one of every 1000 block header for sanity check
-                if (pindexNew->nHeight % 1000 == 0 &&
-                        block_hash != consensusParams.hashGenesisBlock &&
-                        !CheckProof(pindexNew->GetBlockHeader(), consensusParams)) {
-                    return error("%s: CheckProof: %s, %s", __func__, block_hash.ToString(), pindexNew->ToString());
+
+                n_total++;
+                if (diskindex.nHeight >= trim_below_height) {
+                    n_untrimmed++;
+                    pindexNew->proof          = diskindex.proof;
+                    pindexNew->m_dynafed_params       = diskindex.m_dynafed_params;
+                    pindexNew->m_signblock_witness = diskindex.m_signblock_witness;
+
+                    const uint256 block_hash = pindexNew->GetBlockHash();
```

Comment:
another question: is there any formatter to use?

Reply:
Hmmm, I'm not sure what happened here. I believe the old formatting was probably done by hand. It may be that I did a search-and-replace here and didn't notice it got messed up.

Reply:
Fixed formatting.